testkit/src/main/scala/org/bitcoins/testkit/wallet/WalletTestUtil.scala

```diff
@@ -64,5 +72,64 @@ object WalletTestUtil {
   lazy val sampleVout: UInt32 = UInt32.zero
   lazy val sampleSPK: ScriptPubKey =
     ScriptPubKey.fromAsmBytes(hex"001401b2ac67587e4b603bb3ad709a8102c30113892d")
+
   lazy val sampleScriptWitness: ScriptWitness = P2WPKHWitnessV0(freshXpub.key)
+
+  /**
+    * Inserts a incoming TX, and returns it with the address it was sent to
+    *
+    * This method also does some asserts on the result, to make sure what
+    * we're writing and reading matches up
+    */
+  def insertIncomingTx(daos: WalletDAOs)(implicit ec: ExecutionContext): Future[
+    (IncomingTransaction, AddressDb)] = {
+    val WalletDAOs(accountDAO, addressDAO, txDAO, _, _) = daos
+
+    /** Get a TX with outputs */
+    def getTx: Transaction =
+      TransactionGenerators.transaction
```

Comment:
It would probably be worth making a `TransactionGenerators.nonEmptyOutputs` generator